### PR TITLE
[ci] fix some shellcheck errors in scripts

### PR DESCRIPTION
**Pull request by shreyash2002**
Contributes to #6498

- fixed SC0286 (info): Double quote to prevent globbing and word splitting

### CODE REVIEW

**Comment by shreyash2002** [comment]
@microsoft-github-policy-service agree

**Comment by shreyash2002** [comment]
hi @jameslamb 
i've resolved shellchecks check error in three files. all of them had SC0286 check error.

File: .ci/install-clang-devel.sh [hunk 1]
```diff
@@ -32,26 +32,26 @@ add-apt-repository -y "deb-src http://apt.llvm.org/unstable/ llvm-toolchain-${CL
...
-    llvm-${CLANG_VERSION}-dev \
-    llvm-${CLANG_VERSION}-tools \
-    libomp-${CLANG_VERSION}-dev \
-    libc++-${CLANG_VERSION}-dev \
-    libc++abi-${CLANG_VERSION}-dev \
-    libclang-common-${CLANG_VERSION}-dev \
-    libclang-${CLANG_VERSION}-dev \
-    libclang-cpp${CLANG_VERSION}-dev \
-    libunwind-${CLANG_VERSION}-dev
+    clang-"${CLANG_VERSION}" \
```
Comment (jameslamb) [inline .ci/install-clang-devel.sh:35]:
```suggestion
    "clang-${CLANG_VERSION}" \
```

Everywhere in this pull request where you've put double quotes in the middle of a word, please instead only used them around the entire word, like this example.

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute to LightGBM!

I've requested some changes here. I've also made some edits:

* removed `fixed #6498` from the PR title... this does not FIX that issue, it just helps make some progress towards it
* added a link to #6498 in the PR description... please always do this when you contribute here and your contribution relates to other discussions in the project

File: .ci/install-clang-devel.sh (hunk 1 above)
Comment (shreyash2002) [reply .ci/install-clang-devel.sh]: Sure, I've changed the double quotes in .ci/install_clang_devel.sh file.

File: .ci/install-clang-devel.sh [hunk 2]
```diff
@@ -32,26 +32,26 @@ add-apt-repository -y "deb-src http://apt.llvm.org/unstable/ llvm-toolchain-${CL
...
+    "lld-${CLANG_VERSION}" \
+    "llvm-${CLANG_VERSION}"-dev \
+    "llvm-${CLANG_VERSION}"-tools \
+    "libomp-${CLANG_VERSION}"-dev \
+    "libc++-${CLANG_VERSION}"-dev \
+    "libc++abi-${CLANG_VERSION}"-dev \
+    "libclang-common-${CLANG_VERSION}"-dev \
+    "libclang-${CLANG_VERSION}"-dev \
+    "libclang-cpp${CLANG_VERSION}"-dev \
+    "libunwind-${CLANG_VERSION}"-dev
```
Comment (jameslamb) [inline .ci/install-clang-devel.sh:50]:
```suggestion
    "llvm-${CLANG_VERSION}-dev" \
    "llvm-${CLANG_VERSION}-tools" \
    "libomp-${CLANG_VERSION}-dev" \
    "libc++-${CLANG_VERSION}-dev" \
    "libc++abi-${CLANG_VERSION}-dev" \
    "libclang-common-${CLANG_VERSION}-dev" \
    "libclang-${CLANG_VERSION}-dev" \
    "libclang-cpp${CLANG_VERSION}-dev" \
    "libunwind-${CLANG_VERSION}-dev"
```

Please don't leave quotes in the middle of these words, as I asked in https://github.com/microsoft/LightGBM/pull/6504#discussion_r1651475750.

File: .ci/install-clang-devel.sh [hunk 3]
```diff
@@ -32,26 +32,26 @@ add-apt-repository -y "deb-src http://apt.llvm.org/unstable/ llvm-toolchain-${CL
...
 
 # overwriting the stuff in /usr/bin is simpler and more reliable than
 # updating PATH, LD_LIBRARY_PATH, etc.
-cp --remove-destination /usr/lib/llvm-${CLANG_VERSION}/bin/* /usr/bin/
+cp --remove-destination /usr/lib/llvm-"${CLANG_VERSION}"/bin/* /usr/bin/
```
Comment (jameslamb) [inline .ci/install-clang-devel.sh:54]:
```suggestion
cp --remove-destination "$(echo /usr/lib/llvm-${CLANG_VERSION}/bin/*)" /usr/bin/
```

File: .ci/trigger_dispatch_run.sh [hunk 4]
```diff
@@ -37,9 +37,9 @@ pr=$(
 data=$(
   jq -n \
     --arg event_type "$dispatch_name" \
-    --arg pr_number "$(echo $pr | jq '.number')" \
-    --arg pr_sha "$(echo $pr | jq '.head.sha')" \
-    --arg pr_branch "$(echo $pr | jq '.head.ref')" \
+    --arg pr_number "$(echo "$pr" | jq '.number')" \
+    --arg pr_sha "$(echo "$pr" | jq '.head.sha')" \
+    --arg pr_branch "$(echo "$pr" | jq '.head.ref')" \
```
Comment (jameslamb) [inline .ci/trigger_dispatch_run.sh:42]: Putting double quotes inside other double quotes like this requires escaping. To avoid that, please revert these changes and instead add the following comment on line 37 (directly above `data=$(`):

```bash
# shellcheck disable=SC0286
```

**Review by jameslamb — CHANGES_REQUESTED** [review]
thanks for your help, please see my recent suggestions

**Comment by jameslamb** [comment]
@shreyash2002 are you planning to return to this pull request? Do you need any help?

**Comment by StrikerRUS** [comment]
`.ci/install-clang-devel.sh` was removed, `.ci/set_commit_status.sh` and `.ci/trigger_dispatch_run.sh` were fixed in #6621.

@shreyash2002 Thanks for your work here! Please pick up any other issue to contribute if you wish.
